packages/core/admin/admin/src/content-manager/hooks/useLazyComponents/index.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+import { useCustomFields } from '@strapi/helper-plugin';
+
+/**
+ * @description
+ * A hook to lazy load custom field components
+ * @param {Array.<string>} componentUids - The uids to look up components
+ * @returns object
+ */
+const useLazyComponents = (componentUids) => {
+  const [lazyComponentStore, setLazyComponentStore] = useState({});
+  const [loading, setLoading] = useState(true);
```

Comment:
I think we need some error handling.

Maybe we could make this a status string? like the status from react-query hooks. This way status is either 'loading', 'error' or 'success'. It's usually easier to manage than several booleans